### PR TITLE
POSIX: Implement pthread_condattr functions

### LINKED ISSUE #59947: posix: implement `pthread_condattr_getclock()`
There are currently 20 missing POSIX functions with trivial implementations, `pthread_condattr_getclock()` is one of them.

> pthread_condattr_getclock, pthread_condattr_setclock - get and set the clock selection condition variable attribute
> 
> The pthread_condattr_getclock() function shall obtain the value of the clock attribute from the attributes object referenced by attr.
> 
> The pthread_condattr_setclock() function shall set the clock attribute in an initialized attributes object referenced by attr. If pthread_condattr_setclock() is called with a clock_id argument that refers to a CPU-time clock, the call shall fail.
> 
> The clock attribute is the clock ID of the clock that shall be used to measure the timeout service of [pthread_cond_timedwait()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_cond_timedwait.html). The default value of the clock attribute shall refer to the system clock.
> 
> The behavior is undefined if the value specified by the attr argument to pthread_condattr_getclock() or pthread_condattr_setclock() does not refer to an initialized condition variable attributes object.

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_condattr_getclock.html

See RFC #51211 for more info.

### PR DESCRIPTION
**Pull request by Durchbruchswagen**
Added:
pthread_condattr_init
pthread_condattr_destroy
pthread_condattr_getclock
pthread_condattr_setclock

Fixes #59948
Fixes #59947

### CODE REVIEW

File: lib/posix/cond.c [hunk 1]
```diff
@@ -202,4 +202,39 @@ static int pthread_cond_pool_init(void)
 
 	return 0;
 }
+
+int pthread_condattr_init(pthread_condattr_t *att)
+{
+	__ASSERT_NO_MSG(att != NULL);
+
+	att->clock = CLOCK_REALTIME;
```
Comment (cfriedt) [inline lib/posix/cond.c:210]: I would recommend this is set to `CLOCK_MONOTONIC` by default.

**Review by cfriedt — COMMENTED** [review]
Looks good. Just one code change and also please add tests in a separate second commit.

File: lib/posix/cond.c (hunk 1 above)
Comment (Durchbruchswagen) [reply lib/posix/cond.c]: Done, changed default value to clock monotonic

File: tests/posix/common/src/pthread.c [hunk 2]
```diff
@@ -233,6 +233,33 @@ void *thread_top_term(void *p1)
...
+{
+	int ret;
+	clockid_t clock_id;
+	pthread_condattr_t att;
+
+	ret = pthread_condattr_init(&att);
+	zassert_equal(ret, 0, "pthread_condattr_init failed");
+
+	ret = pthread_condattr_getclock(&att, &clock_id);
+	zassert_equal(ret, 0, "pthread_condattr_getclock failed");
```
Comment (cfriedt) [inline tests/posix/common/src/pthread.c:246]: Would be good to apply this pattern elsewhere
```suggestion
	zassert_ok(pthread_condattr_getclock(&att, &clock_id));
```

Comment (Durchbruchswagen) [reply tests/posix/common/src/pthread.c]: Done, replaced `zassert_equal` with `zassert_ok` wherever i could and removed `ret` variable

File: tests/posix/common/src/pthread.c [hunk 3]
```diff
@@ -233,6 +233,28 @@ void *thread_top_term(void *p1)
 	return NULL;
 }
 
+ZTEST(posix_apis, test_pthread_condattr)
```
Comment (cfriedt) [inline tests/posix/common/src/pthread.c:236]: If you end up having to respin this for whatever reason, this should probably go into cond.c

**Review by cfriedt — APPROVED** [review]
LGTM - thanks for your contribution @Durchbruchswagen 👍 

Just in case there is another respin for whatever reason, could move the test to `cond.c`

**Review by mbolivar-ampere — APPROVED** [review]

**Pull request merged**